code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -359,24 +361,56 @@
 		P.fire()
 		return P
 
-/mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
-	return iswallturf(T) || ismineralturf(T)
 
-/mob/living/simple_animal/hostile/proc/DestroySurroundings()
+/mob/living/simple_animal/hostile/proc/DestroyStructuresCycle(turf/T)
+	var/dir_to_target = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
+	var/list/dir_list = list(dir_to_target, turn(dir_to_target, 90), turn(dir_to_target, 270), turn(dir_to_target, 180))
+	cycle_loop:
+		for(var/dir in dir_list)
+			T = get_step(targets_from, dir)
+			for(var/a in T)
+				var/atom/A = a
+				if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
+					A.attack_animal(src)
+					break cycle_loop
```

Comment:
you could just return instead of using a named loop.
also, I'd do this like the following, which is probably clearer and more like what I'm expecting the code to do;
````
	var/dir_to_target = get_cardinal_step_towards(targets_from, target)
	var/dir_list = list()
	if(dir_to_target in GLOB.diagonals) //it's diagonal, so we need two directions to hit
		for(var/direction in GLOB.cardinals)
			if(direction & dir_to_target)
				dir_list += direction
	else
		dir_list += dir_to_target
	var/turf/T
	for(var/direction in dir_list) //now we hit all of the directions we got in this fashion, since it's the only directions we should actually need
		T = get_step(targets_from, direction)
		if(CanSmashTurfs(T))
			T.attack_animal(src)
		for(var/a in T)
			var/atom/A = a
			if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
				A.attack_animal(src) //we neither break nor return here, so that it hits both directions at once instead of possibly being inefficient. this is still better than hitting every possible turf at once, since it'll be aimed at the target.
````

Reply:
i know its really a subjective choice in the end but i think its more realistic that a hostile mob would only attack one turf at a time
while hitting 2 turfs at a time can fix the diagonal issue cleanly and stuff, it doesnt make sense since it would be the only proc that allows mobs to do that consistently

Reply:
hitting both is cleaner in the case where one is easier to destroy than the other, but would be attacked second, and is thus better.

Reply:
couldnt you just check to see which of the 2 is easier to destroy instead of hitting them both

Reply:
that's difficult to determine.

Reply:
couldnt you just make them hit the walls first then hit the structures with lower integrity first